src/content_sao.h

```diff
@@ -322,6 +322,16 @@ class PlayerSAO : public UnitSAO
 	{
 		return m_dig_pool;
 	}
+	void setMaxSpeedOverride(const v3f &vel)
+	{
+		if (m_max_speed_override_time == 0.0f)
+			m_max_speed_override = vel;
+		else
+			m_max_speed_override += vel;
+		// we hope the player will hit a wall or something within 2 seconds
+		// FIXME improve this somehow
+		m_max_speed_override_time = 2.0f;
```

Comment:
```C++
m_client->max_speed_override_time = 2 * vel / MYMIN(
	m_client->movement_acceleration_default, m_client->movement_acceleration_air);
```
Plus maybe some little tolerance for delayed packets.

Reply:
why the `2 *`?
shouldn't that be `MYMAX`?

Reply:
`2 *` is an error (`vel` is the max speed, not average over that period).
`MYMIN` because you want the bigger override time, depending on what's slowing you down less.